activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/query/rest/ProcessInstanceVariableController.java

```diff
@@ -49,9 +55,19 @@ public ProcessInstanceVariableController(VariableResourceAssembler variableResou
 
     @RequestMapping(value = "/variables", method = RequestMethod.GET)
     public PagedResources<VariableResource> getVariables(@PathVariable String processInstanceId,
+                                                         @QuerydslPredicate(root = Variable.class) Predicate predicate,
                                                          Pageable pageable,
                                                          PagedResourcesAssembler<Variable> pagedResourcesAssembler) {
-        Page<Variable> variables = variableRepository.findAll(QVariable.variable.processInstanceId.eq(processInstanceId), pageable);
+
+        QVariable variable = QVariable.variable;
+        BooleanExpression expression = variable.processInstanceId.eq(processInstanceId);
+
+        if(predicate != null){
```

Comment:
Missing else condition. If the predicate is null we still want to apply the `expression` condition.

Reply:
Fixed with the last commit.